alacritty/src/daemon.rs

```diff
@@ -1,34 +1,53 @@
 use std::ffi::OsStr;
+use std::fmt::Debug;
 use std::io;
-use std::process::{Command, Stdio};
-
 #[cfg(not(windows))]
 use std::os::unix::process::CommandExt;
-
 #[cfg(windows)]
 use std::os::windows::process::CommandExt;
+use std::process::{Command, Stdio};
+
+use log::{debug, warn};
+
 #[cfg(windows)]
 use winapi::um::winbase::{CREATE_NEW_PROCESS_GROUP, CREATE_NO_WINDOW};
 
-/// Threading utilities.
-pub mod thread {
-    /// Like `thread::spawn`, but with a `name` argument.
-    pub fn spawn_named<F, T, S>(name: S, f: F) -> std::thread::JoinHandle<T>
-    where
-        F: FnOnce() -> T + Send + 'static,
-        T: Send + 'static,
-        S: Into<String>,
-    {
-        std::thread::Builder::new().name(name.into()).spawn(f).expect("thread spawn works")
-    }
+/// Start the daemon and log error on failure.
+pub fn start_daemon<I, S>(program: &str, args: I)
+where
+    I: IntoIterator<Item = S> + Debug + Copy,
+    S: AsRef<OsStr>,
+{
+    match start_daemon_impl(program, args) {
+        Ok(_) => debug!("Launched {} with args {:?}", program, args),
+        Err(_) => warn!("Unable to launch {} with args {:?}", program, args),
+    };
+}
 
-    pub use std::thread::*;
+#[cfg(windows)]
+fn start_daemon_impl<I, S>(program: &str, args: I) -> io::Result<()>
```

Comment:
```suggestion
fn spawn_daemon<I, S>(program: &str, args: I) -> io::Result<()>
```

I think adding `_impl` is a bit strange. A similar but not identical name should do.